lib/Test/PostgreSQL.pm

```diff
@@ -705,6 +730,13 @@ Stops postmaster.
 
 Setups the PostgreSQL instance.
 
+=head2 use_socket
+
+Whether to only connect via sockets; if false (the default),
+connections can occur via localhost. [This changes the L</dsn>
+returned to only give the socket directory, and avoids any issues with
+conflicting TCP ports on localhost.]
```

Comment:
This language needs to change. Perhaps you don't realise, but TCP connections are also called sockets!

You can have both "TCP Sockets" and "UNIX sockets".

I think this comment needs to explicitly mention that its using the UNIX file-socket.

Reply:
Sure; would it be OK to just say UNIX sockets?

Reply:
Yes, but can you change the method name from `use_socket` to `unix_socket` or `enable_unix_socket` or something like that?

Reply:
Sure, which would you prefer?

Reply:
I don't have strong feelings, but... `unix_socket` ?
So when you're setting up the library it'll look like:
`Test::PostgreSQL->new(unix_socket => true, socket_dir => "/my/dir")`

Although speaking of the socket_dir -- could you make it default to `File::Temp::tempdir` or similar, with the CLEANUP flag enabled? Otherwise tests run in parallel will clash.

Reply:
Sure, `unix_socket` works. To use, you just do:

  Test::PostgreSQL->new(unix_socket => 1);

because the code base already creates the socket directory and removes it. I just abstracted out socket_dir instead of doing `File::Spec->catdir($self->base_dir,'tmp')` every time we needed it, and to make it clearer that that's what the socket directory actually is.